src/transformers/generation_tf_utils.py

```diff
@@ -650,7 +650,11 @@ def generate(
 
         # current position and vocab size
         cur_len = shape_list(input_ids)[1]  # unused
-        vocab_size = self.config.vocab_size
+        vocab_size = getattr(self.config, "vocab_size", None)
+        if vocab_size is None and self.config.is_encoder_decoder:
+            decoder_config = getattr(self.config, "decoder", None)
+            if decoder_config is not None:
+                vocab_size = getattr(self.config.decoder, "vocab_size", None)
```

Comment:
- `EncoderDecoderConfig` doesn't have `vocab_size`.
- `RagConfig` has `None` as `vocab_size`, but not has `decoder` attribute.

Maybe it's better to set `EncoderDecoderConfig.vocab_size` to `None`, and  replace the 1st line by
`vocab_size = self.config.vocab_size`.